spec/models/section_spec.rb

```diff
@@ -142,6 +142,24 @@ def key_classes_for(hash)
     end
   end
 
+  describe '#save' do
+    it 'saves the last two editions' do
+      slug_generator = double(:slug_generator)
+      editions = [
+        edition_1 = double(:edition_1),
+        edition_2 = double(:edition_2),
+        edition_3 = double(:edition_3)
+      ]
+      section = Section.new(slug_generator, 'section-id', editions)
+
+      expect(edition_1).to_not receive(:save!)
+      expect(edition_2).to receive(:save!)
+      expect(edition_3).to receive(:save!)
```

Comment:
OK. I've added #1016 to capture this possible problem, because I'm concerned that the code my only be working by accident and we probably won't have any other trigger to investigate it.